File: aplication/client/FE/src/app/shared/model/entity/Payment.ts
import {Booking} from './Booking';
import {JsonProperty} from 'jsog-typescript';
import {MovieShowtime} from './MovieShowtime';

export class Payment {
  id: number;
  name: string;
  bookings: Booking[];

  constructor(id: number, name: string, bookings: Booking[]) {
    this.id = id;
    this.name = name;
    this.bookings = bookings;
  }
}

File: aplication/server/BE/src/main/java/com/model/entity/Payment.java
package com.model.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.voodoodyne.jackson.jsog.JSOGGenerator;

import javax.persistence.*;
import java.util.List;
@Entity

public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    @OneToMany(mappedBy = "payment", cascade = CascadeType.ALL)
    @JsonBackReference
    private List<Booking> bookings;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Booking> getBookings() {
        return bookings;
    }

    public void setBookings(List<Booking> bookings) {
        this.bookings = bookings;
    }
}
